### PR TITLE
Update Java SDK

### PR DESCRIPTION
**Pull request by stephanos**
## What was changed

Updated Java SDK to pull in new Update-with-Start API.

### CODE REVIEW

File: core/src/main/java/io/temporal/samples/earlyreturn/EarlyReturnClient.java [hunk 1]
```diff
@@ -49,17 +49,13 @@ private static void runWorkflowWithUpdateWithStart(WorkflowClient client) {
...
     TxResult updateResult = null;
     try {
-      WorkflowUpdateHandle<TxResult> updateHandle =
-          WorkflowClient.updateWithStart(workflow::processTransaction, txRequest, updateOp);
-
-      updateResult = updateHandle.getResultAsync().get();
+      updateResult =
+          WorkflowClient.executeUpdateWithStart(
+              workflow::returnInitResult,
+              UpdateOptions.<TxResult>newBuilder().build(),
```
Comment (stephanos) [inline core/src/main/java/io/temporal/samples/earlyreturn/EarlyReturnClient.java:57]: @dandavison 😬

File: core/src/main/java/io/temporal/samples/countinterceptor/SimpleCountWorkerInterceptor.java [hunk 2]
```diff
@@ -34,4 +36,10 @@ public WorkflowInboundCallsInterceptor interceptWorkflow(WorkflowInboundCallsInt
   public ActivityInboundCallsInterceptor interceptActivity(ActivityInboundCallsInterceptor next) {
     return new SimpleCountActivityInboundCallsInterceptor(next);
   }
+
+  @Override
+  public NexusOperationInboundCallsInterceptor interceptNexusOperation(
```
Comment (Quinn-With-Two-Ns) [inline core/src/main/java/io/temporal/samples/countinterceptor/SimpleCountWorkerInterceptor.java:41]: The "correct" solution is these samples should be `SimpleCountWorkerInterceptor extends BaseWorkerInterceptor `

File: core/src/main/java/io/temporal/samples/countinterceptor/SimpleCountWorkerInterceptor.java [hunk 3]
```diff
@@ -19,11 +19,11 @@
...
 
-import io.temporal.common.interceptors.ActivityInboundCallsInterceptor;
-import io.temporal.common.interceptors.WorkerInterceptor;
-import io.temporal.common.interceptors.WorkflowInboundCallsInterceptor;
+import io.nexusrpc.handler.OperationContext;
+import io.temporal.common.interceptors.*;
+import io.temporal.internal.sync.BaseRootWorkflowInboundCallsInterceptor;
 
-public class SimpleCountWorkerInterceptor implements WorkerInterceptor {
+public class SimpleCountWorkerInterceptor extends WorkerInterceptorBase {
```
Comment (stephanos) [inline core/src/main/java/io/temporal/samples/countinterceptor/SimpleCountWorkerInterceptor.java:26]: To provide a base implementation for Nexus.

File: core/src/test/java/io/temporal/samples/safemessagepassing/ClusterManagerWorkflowWorkerTest.java [hunk 4]
```diff
@@ -105,9 +105,8 @@ public void testUpdateIdempotency() {
             .newWorkflowStub(
                 ClusterManagerWorkflow.class,
                 WorkflowOptions.newBuilder().setTaskQueue(testWorkflowRule.getTaskQueue()).build());
-    CompletableFuture<ClusterManagerWorkflow.ClusterManagerResult> result =
```
Comment (stephanos) [inline core/src/test/java/io/temporal/samples/safemessagepassing/ClusterManagerWorkflowWorkerTest.java:108]: I got an error for an unused variable 🤷

File: core/src/main/java/io/temporal/samples/earlyreturn/EarlyReturnClient.java (hunk 1 above)
Comment (steveandroulakis) [reply core/src/main/java/io/temporal/samples/earlyreturn/EarlyReturnClient.java:58]: Much cleaner implementation!

**Review by steveandroulakis — APPROVED** [review]
Early return client code updates look good

**Pull request merged**